### DISCUSSION TITLE
Does a sidekiq job have a runtime limit?

### DISCUSSION THREAD

**Discussion by jai-x** [sidekiq/sidekiq#6210] [Q&A]
Is there any system limitation that would prevent a job from running forever?

```ruby
class InifiniteJob
  include Sidekiq::Worker
  
  def perform
    loop do
      sleep 10
    end
  end
end
```

Assuming this job is run on a Sidekiq worker and there's no disruption to the worker or the backing Redis, is there a limit?

Would there be some sort of int overflow that is non-obvious?

**Comment by mperham** [comment]
Sidekiq is designed for short-lived jobs. It's possible it would work but it's not an officially supported usecase and some functionality wouldn't work well with that design. It's better to have a cron job create a new job every minute and do N seconds of processing in each job. Sidekiq does have a 30 second shutdown timeout so jobs which run longer can slow down your deploys.
